pyanaconda/modules/common/structures/partitioning.py

```diff
@@ -18,10 +18,232 @@
+        """Should devices be encrypted?
+
+        :return: True or False
+        """
+        return self._encrypted
+
+    @encrypted.setter
+    def encrypted(self, encrypted: Bool):
+        self._encrypted = encrypted
+
+    @property
+    def passphrase(self) -> Str:
+        """Passphrase for all encrypted devices.
+
+        :return: a string with the passphrase
+        """
+        return self._passphrase
+
+    @passphrase.setter
+    def passphrase(self, value: Str):
+        self._passphrase = value
+
+    @property
+    def cipher(self) -> Str:
+        """Encryption algorithm used to encrypt the filesystem.
+
+        For example: aes-xts-plain64
+
+        :return: a name of an algorithm
+        """
+        return self._cipher
+
+    @cipher.setter
+    def cipher(self, cipher: Str):
+        self._cipher = cipher
+
+    @property
+    def luks_version(self) -> Str:
+        """Version of LUKS.
+
+        Example: 'luks2'
+
+        :return: a string with the LUKS version
+        """
+        return self._luks_version
+
+    @luks_version.setter
+    def luks_version(self, version: Str):
+        self._luks_version = version
+
+    @property
+    def pbkdf(self) -> Str:
+        """The PBKDF algorithm.
+
+        Set Password-Based Key Derivation Function (PBKDF)
+        algorithm for LUKS keyslot.
+
+        Example: 'argon2i'
+
+        :return: a name of the algorithm
+        """
+        return self._pbkdf
+
+    @pbkdf.setter
+    def pbkdf(self, pbkdf: Str):
+        self._pbkdf = pbkdf
+
+    @property
+    def pbkdf_memory(self) -> Int:
+        """The memory cost for PBKDF."""
+        return self._pbkdf_memory
+
+    @pbkdf_memory.setter
+    def pbkdf_memory(self, memory: Int):
+        """Set the memory cost for PBKDF.
+
+        :param memory: the memory cost in kilobytes
+        """
+        self._pbkdf_memory = memory
+
+    @property
+    def pbkdf_time(self) -> Int:
+        """The time to spend with PBKDF processing.
+
+        Sets the number of milliseconds to spend with PBKDF
+        passphrase processing.
+
+        :return: a number of milliseconds
+        """
+        return self._pbkdf_time
+
+    @pbkdf_time.setter
+    def pbkdf_time(self, time_ms: Int):
+        self._pbkdf_time = time_ms
+
+    @property
+    def pbkdf_iterations(self) -> Int:
+        """The number of iterations for PBKDF.
+
+        Avoid PBKDF benchmark and set time cost (iterations) directly.
+
+        :return: a number of iterations
+        """
+        return self._pbkdf_iterations
+
+    @pbkdf_iterations.setter
+    def pbkdf_iterations(self, iterations: Int):
+        self._pbkdf_iterations = iterations
+
+    @property
+    def escrow_certificate(self) -> Str:
+        """URL of an X.509 certificate.
+
+        Store the data encryption keys of all encrypted volumes created during
+        installation, encrypted using the certificate, as files in /root.
+
+        :return: URL of an X.509 certificate
```

Comment:
is this a file:// URL or regular URL, eq.:

`file://foo.cert`

or

`/foo.cert`

Reply:
I don't know.